plugin/documents.py

```diff
@@ -544,9 +544,11 @@ def _on_complete_result(self, response: Optional[Union[dict, List]], completion_
                             can_resolve_completion_items: bool) -> None:
         response_items = []  # type: List[Dict]
         flags = 0
-        if userprefs().only_show_lsp_completions:
-            flags |= sublime.INHIBIT_WORD_COMPLETIONS
+        prefs = userprefs()
+        if prefs.inhibit_explicit_completions:
             flags |= sublime.INHIBIT_EXPLICIT_COMPLETIONS
+        if prefs.inhibit_word_completions:
+            flags |= sublime.INHIBIT_WORD_COMPLETIONS
             flags |= sublime.INHIBIT_REORDER
```

Comment:
What's the motivation for not using INHIBIT_REORDER when inhibiting explicit completions?